DQMServices/Core/interface/DQMEDHarvester.h

```diff
@@ -24,13 +26,14 @@ class DQMEDHarvester
   // implicit assignment operator
   // implicit destructor
   virtual void beginRun(edm::Run const&, edm::EventSetup const&) {};
-  virtual void analyze(edm::Event const&, edm::EventSetup const&) final {};
+  virtual void produce(edm::Event&, edm::EventSetup const&) override final {};
```

Comment:
All of the functions which just have an implementation of `{}` and which are not marked as `final` can have those default behavior functions removed from the header. A class which inherits from `EDHarvester` would still be able to override those functions.